lib/vtls/bearssl.c

```diff
@@ -339,6 +339,8 @@ static CURLcode bearssl_connect_step1(struct Curl_easy *data,
   struct in_addr addr;
 #endif
 
+  DEBUGASSERT(backend != NULL);
```

Comment:
I think you should remove `!= NULL` from these checks, as we normally check pointers for being non-NULL like that everywhere else.